### PR TITLE
Use same method to get stake and regular txs

### PR DESCRIPTION
**Pull request by vctt94**
~~This PR depends on #2456~~ #2456 was closed to maintain only one PR

It is a continuation of the idea proposed at #2456.

@matheusd thought it would be better creating a POC before going forward with this idea.

Right now this are the changes made so far at this PR:

- selectors: added transactionsMap, regularTransactions and stakeTransactions to separate logic between them, which one is used to keep record of all fetched txs, one to show at transactions history page, the other to show at My Tickets Page. Removed many selectors.
- Actions: 
    - ClientActions: Moved many methods to new TransactionActions.
    - DecodedMessageActions: Removed. We are now decoding all transactions locally.
        - Added a way to decode scripts locally, with new app/helpers/scripts.js.
        - Added a way to selializeNoWitnessEncode, so we can sign trezor txs.
    - TransactionActions: Removed method getTickets. Using only getTransactions to get all transactions and using the new selectors for categorize them.
    - TrezorActions: Moved some aux methods to helpers, changed how decode transactions for local method.
         - For that we need a way to selializeNoWitnessEncode, because we need to generate the hash for unsignedTxs.
- Components: Removed TxDetails components, change transactions page to functional component, multi other small changes.
- connectors: Removed transactionDetails and transactionPage connectors, multi other small changes in others.
- constants: Added some needed Decred chainparams constants and OP codes for new methods, created trezor.js, moved some constants to decrediton.
- helpers: Added new helpers, script.js for extracting address from raw script, msgTx for decodingRawTx and selializeNoWitnessEncode and trezor.js, which I moved some aux methods from TrezorActions. Some other multi small changes
- reducers: doing some clean up and necessary changes.
- tests: Adding unit test for decodeRawTx and new helpers.

Still left: 

  - ~~Fix filter selectors, when showing txs, so they work properly~~.
  - ~~Fix duplicated keys at MyTickets Page~~
  - ~~Right after buying a ticket, decrediton is crashing~~
  - ~~Going to a pending tx crashes decrediton~~
  -  ~~Fix the bug described at #2456, which is the cause of all that change :p~~

### CODE REVIEW

File: app/actions/DecodeMessageActions.js [hunk 1]
```diff
@@ -20,28 +20,39 @@ export const getDecodeMessageServiceAttempt = () => (dispatch, getState) => {
...
-    .all(hexTxs.map(hex => decodeTransaction(decodeMessageService, hex)))
-    .then(resolved)
-    .catch(error => dispatch({ error, type: DECODERAWTXS_FAILED }));
-};
-
 // decodeRawTransaction requests decodification of a raw, hex-encoded transaction.
-export const decodeRawTransaction = (hexTx) => decodeRawTransactions([ hexTx ]);
+export const decodeRawTransaction = (hexTx) => (dispatch, getState) => new Promise((resolve, reject) => {
+  const { decodeMessageService } = getState().grpc;
+  decodeTransaction(decodeMessageService, hexTx)
```
[the remark this answers is not in the document]
Comment (vctt94) [reply app/actions/DecodeMessageActions.js:26]: I like this idea. Having two ways of decoding a raw transaction is kind of messy

[the remark this answers is not in the document]
Comment (alexlyp) [reply app/actions/DecodeMessageActions.js:26]: Yup agree, we should just be using decodeRawTransaction currently.

File: app/wallet/service.js [hunk 2]
```diff
@@ -313,6 +318,223 @@ export const decodeRawTransaction = (rawTx, chainParams) => {
   tx.expiry = rawTx.readUInt32LE(position);
   position += 4;
 
-  console.log(tx)
   return tx;
 };
+
+// selializeNoWitnesseEncode gets a decoded tx and serialize it.
```
Comment (matheusd) [inline app/wallet/service.js:324]: Extra `e` (should have been `serializeNoWitnessEncode`)

**Comment by matheusd** [comment]
Actually,   another small UI nit  but it can go on a future PR: 

- On vote transactions, the previous (voted on) block and vote bits show up as a blue line. We probably want to process them in a special way and show the hash of the block and the decoded vote bits (agendas voted, whether the block was approved or not)

**Comment by alexlyp** [comment]
Needs a rebase, then should be all set.

**Pull request merged**